src/configProvider/index.js

```diff
@@ -1,65 +1,42 @@
 // @flow
 import * as React from 'react'
-import PropTypes from 'prop-types'
-import type {
-  ConfigProviderContext,
-  DisplayAliases,
-  SpacingAliases,
-} from '../types'
-import defaults from '../defaults'
+import type { ConfigContextType } from '../types'
+import Context from './context'
 
-type Props = {
-  base?: number,
+type ProviderProps = {
+  ...ConfigContextType,
   children?: React.Node,
-  columns?: number,
-  displayAliases?: DisplayAliases,
-  fallbackDisplayKey?: string,
-  gutter?: number,
-  maxPageWidth?: number | 'none',
-  minPageWidth?: number,
-  pageMargin?: {
-    [string]: number,
-  },
-  spacingAliases?: SpacingAliases,
-  verticalGutter?: number,
 }
 
-export const ConfigContextPropTypes = {
-  gymnast: PropTypes.shape({
-    base: PropTypes.number,
-    columns: PropTypes.number,
-    displayAliases: PropTypes.shape({}),
-    fallbackDisplayKey: PropTypes.string,
-    gutter: PropTypes.number,
-    maxPageWidth: PropTypes.oneOf([PropTypes.number, 'none']),
-    minPageWidth: PropTypes.number,
-    pageMargin: PropTypes.shape({}),
-    spacingAliases: PropTypes.shape({}),
-    verticalGutter: PropTypes.number,
-  }),
-}
-
-export default class ConfigProvider extends React.Component<Props> {
-  static contextTypes = ConfigContextPropTypes
-  static childContextTypes = ConfigContextPropTypes
+type ProviderState = ConfigContextType | {}
 
-  getChildContext(): ConfigProviderContext {
-    const {
-      gutter = defaults.gutter,
-      verticalGutter = gutter,
-      children,
-      ...props
-    } = this.props
+// This HOC takes a Provider component and wraps it around a Consumer
+// component that provides non-defaulted values that have been set from preceding
+// Provider components.
+class ConfigProvider extends React.Component<ProviderProps, ProviderState> {
+  static getDerivedStateFromProps(props: ProviderProps, state: ProviderState) {
+    const { children, ...restProps } = props
 
     return {
-      gymnast: {
-        gutter,
-        verticalGutter,
-        ...props,
-      },
+      ...state,
+      ...restProps,
     }
   }
+
+  state = {}
```

Comment:
Decided to keep this here, since getting warning via `yarn test`:

```
$ jest -u
 PASS  src/asCol/asCol.spec.js
 PASS  src/asLayout/layout.spec.js
 PASS  src/offset/offset.spec.js
 PASS  src/root/root.spec.js
 PASS  src/dev/dev.spec.js
 PASS  src/withResolution/withResolution.logic.spec.js
 PASS  src/core/core.spec.js
 PASS  src/withResolution/withResolution.spec.js
 PASS  src/log/log.spec.js
 PASS  src/withResolution/mediaQuery.spec.js
 PASS  src/errors/index.spec.js
 PASS  src/utils/utils.spec.js
 PASS  src/asGrid/grid.spec.js
 PASS  src/configProvider/index.spec.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:33
      Warning: ConfigProvider: Did not properly initialize state during construction. Expected state to be an object, but it was undefined.

 PASS  test/e2e/gymnast.spec.js
 PASS  src/withContext/index.spec.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:33
      Warning: ConfigProvider: Did not properly initialize state during construction. Expected state to be an object, but it was undefined.


Test Suites: 16 passed, 16 total
Tests:       3 skipped, 118 passed, 121 total
Snapshots:   4 passed, 4 total
Time:        4.683s
Ran all test suites.
✨  Done in 6.51s.
```